components/authz-service/server/v2/project.go

```diff
@@ -384,6 +384,24 @@ func (s *state) GetRule(ctx context.Context, req *api.GetRuleReq) (*api.GetRuleR
 	return &api.GetRuleResp{Rule: apiRule}, nil
 }
 
+func (s *state) GetStagedRule(ctx context.Context, req *api.GetStagedRuleReq) (*api.GetStagedRuleResp, error) {
+	resp, err := s.store.GetStagedRule(ctx, req.Id)
+	if err != nil {
+		if err == storage_errors.ErrNotFound {
+			return nil, status.Errorf(codes.NotFound, "could not find rule with ID %q", req.Id)
+		}
+		return nil, status.Errorf(codes.Internal,
+			"error retrieving rule with ID %q: %s", req.Id, err.Error())
+	}
+
+	apiRule, err := fromStorageRule(resp)
```

Comment:
Will this work? I would imagine the struct is a bit different bc of the additional field 🤔